spanner/grpc_client.go

```diff
@@ -82,7 +92,22 @@ func newGRPCSpannerClient(ctx context.Context, sc *sessionClient, opts ...option
 		return nil, err
 	}
 
+	// Retrieve the channelID for each spannerClient.
+	// It is assumed that this method is invoked
+	// under a lock already.
+	channelID, ok := sc.channelIDMap[raw]
```

Comment:
We do use it to retain the mapping of vkit.Client to channelIDs; we do need it because between multiple sessionClients there are different vkit.Client objects created from each sessionClient. Is my reasoning and understanding correct?